src/components/POLine/hooks/useGetDonorsByFundId/useGetDonorsByFundId.js

```diff
@@ -0,0 +1,50 @@
+import {
+  get,
+  keyBy,
+  uniq,
+} from 'lodash';
+import {
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+export const useGetDonorsByFundId = ({
+  funds,
+  fundDistribution,
+  initialDonorOrganizationIds,
+}) => {
+  const [donorIds, setDonorIds] = useState(initialDonorOrganizationIds);
+  const [excludedIds, setExcludedIds] = useState([]);
+
+  const fundRecordsMap = useMemo(() => keyBy(funds, 'id'), [funds]);
+  const fundIds = useMemo(() => {
+    return fundDistribution?.map(({ fundId }) => fundId).filter(Boolean);
+  }, [fundDistribution]);
+
+  const donorOrganizationIds = useMemo(() => {
+    return fundIds?.reduce((acc, fundId) => {
+      const fund = fundRecordsMap[fundId];
+      const fundDonorIds = get(fund, 'donorOrganizationIds', []).filter(donorId => !excludedIds.includes(donorId));
+
+      return [...acc, ...fundDonorIds];
```

Comment:
I think it makes sense to mutate the accumulated value to maintain performance.

Reply:
In this case, this is working with a small array of strings and this will not be noticeable, though